packages/scoutgame-ui/src/components/create-project/components/ProjectSmartContractForm.tsx

```diff
@@ -0,0 +1,354 @@
+import { CONTRACT_DEPLOYER_SIGN_MESSAGE } from '@packages/scoutgame/projects/constants';
+import type { CreateScoutProjectFormValues } from '@packages/scoutgame/projects/createScoutProjectSchema';
+import { getContractDeployerAddressAction } from '@packages/scoutgame/projects/getContractDeployerAddressAction';
+import Image from 'next/image';
+import { useAction } from 'next-safe-action/hooks';
+import { useCallback, useMemo, useState } from 'react';
+import { useFieldArray, type Control } from 'react-hook-form';
+import { verifyMessage } from 'viem';
+import { useSignMessage } from 'wagmi';
+
+import { Dialog } from '../../common/Dialog';
+import { FormErrors } from '../../common/FormErrors';
+import { chainRecords } from '../../projects/constants';
+
+export type Deployer = { address: string; verified: boolean; signature: string | null };
+
+export function ProjectSmartContractForm({
+  control,
+  deployers,
+  setDeployers
+}: {
+  control: Control<CreateScoutProjectFormValues>;
+  deployers: Deployer[];
+  setDeployers: React.Dispatch<React.SetStateAction<Deployer[]>>;
+}) {
+  const { executeAsync: getContractDeployerAddress, isExecuting } = useAction(getContractDeployerAddressAction);
+  const [open, setOpen] = useState(false);
+  const [isConfirmModalOpen, setIsConfirmModalOpen] = useState(false);
+  const [selectedContract, setSelectedContract] = useState<`0x${string}` | null>(null);
+
+  const { signMessageAsync } = useSignMessage();
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const {
+    fields: contracts,
+    append,
+    remove,
+    update
+  } = useFieldArray({
+    control,
+    name: 'contracts'
+  });
+  const [tempContract, setTempContract] = useState<{ address: string; chainId: number } | null>(null);
+
+  const verifyDeployerOwnership = useCallback(
+    async (deployerAddress: `0x${string}`) => {
+      try {
+        const signature = await signMessageAsync({
+          message: CONTRACT_DEPLOYER_SIGN_MESSAGE
+        });
+
+        const isValid = await verifyMessage({
+          message: CONTRACT_DEPLOYER_SIGN_MESSAGE,
+          signature,
+          address: deployerAddress
+        });
+
+        return { signature, isValid };
+      } catch (error) {
+        setErrorMessage('Error verifying deployer ownership');
+        log.info('Error verifying deployer ownership', { error });
+        return { signature: null, isValid: false };
+      }
+    },
+    [signMessageAsync]
+  );
+
+  const onCancel = useCallback(() => {
+    setTempContract(null);
+    setOpen(false);
+  }, [setTempContract, setOpen]);
+
+  const onSave = useCallback(async () => {
+    if (tempContract) {
+      const getContractDeployerAddressResult = await getContractDeployerAddress({
+        chainId: Number(tempContract.chainId),
+        contractAddress: tempContract.address
+      });
+
+      const deployerAddress = getContractDeployerAddressResult?.data;
+
+      if (!deployerAddress) {
+        return;
```

Comment:
i tried using my wallet address for a contract, i know its wrong, but when i do that and click Save, nothing happens. i should get an error message someplace